src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs

```diff
@@ -522,9 +591,20 @@ internal static SafeAreaDetails GetInstance(FrameworkElement element)
 			internal void OnInsetsChanged(InsetMask oldValue, InsetMask newValue)
 			{
 				_insetMask = newValue;
+
+				VerifySoftInputUsage();
+
 				UpdateInsets();
 			}
 
+			private void VerifySoftInputUsage()
+			{
+				if (HasSoftInput() && Owner is { } owner && (owner is not SafeArea or ScrollViewer))
+				{
+					_log.WarnIfEnabled(() => "Usage of InsetMask.SoftInput while not attached to a ScrollViewer may result in undesirable behavior");
```

Comment:
Should we point to an article to explain why?

Reply:
Yeah the docs are almost ready for PR :)